environment/src/main/java/com/sequenceiq/environment/environment/service/stack/StackService.java

```diff
@@ -0,0 +1,65 @@
+package com.sequenceiq.environment.environment.service.stack;
+
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.StackV4Endpoint;
+import com.sequenceiq.cloudbreak.cloud.scheduler.PollGroup;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.flow.config.update.config.EnvStackConfigUpdatesFlowConfig;
+import com.sequenceiq.environment.store.EnvironmentInMemoryStateStore;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.service.FlowCancelService;
+import com.sequenceiq.flow.service.flowlog.FlowLogDBService;
+import javax.ws.rs.BadRequestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class StackService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackService.class);
+
+    private final StackV4Endpoint stackV4Endpoint;
+
+    private final FlowCancelService flowCancelService;
+
+    private final FlowLogDBService flowLogDBService;
+
+    public StackService(
+        StackV4Endpoint stackV4Endpoint,
+        FlowCancelService flowCancelService,
+        FlowLogDBService flowLogDBService) {
+        this.stackV4Endpoint = stackV4Endpoint;
+        this.flowCancelService = flowCancelService;
+        this.flowLogDBService = flowLogDBService;
+    }
+
+    public void triggerConfigUpdateForStack(String stackName) {
+        try {
+            stackV4Endpoint.updatePillarConfigurationByName(0L, stackName);
+        } catch (BadRequestException bre) {
+            LOGGER.info(String
+                .format("Unable to start config update for stack %s.  Message is %s", stackName,
+                    bre.getMessage()));
+            throw bre;
+        }
+    }
+
+    public void cancelRunningStackConfigUpdates(Environment environment) {
+        try {
+            FlowLog flowLog = flowLogDBService.getLastFlowLogByResourceCrnOrName(environment.getResourceCrn());
```

Comment:
I think in this case we want it to fetch from the DB again.  Since it is how we are checking for a running flow across multiple instances of the service, we need to check and not depend on a possible slow cache.

Reply:
you misinterpreted what I said. If you check the implementation, `getLastFlowLogByResourceCrnOrName` will fetch the environment again to get it's id. While you already has the environment and could use it's id.

Reply:
k fixed